src/EditorFeatures/Core/EditorConfigSettings/Updater/NamingStyles/NamingStyleSettingsUpdater.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data;
+using Microsoft.CodeAnalysis.EditorConfig.Parsing.NamingStyles;
+using Microsoft.CodeAnalysis.NamingStyles;
+using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.EditorConfig.Parsing.NamingStyles.EditorConfigNamingStylesParser;
+
+namespace Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Updater
+{
+    internal partial class NamingStyleSettingsUpdater : SettingsUpdaterBase<(Action<(object, object?)> onSettingChange, NamingStyleSetting option), object>
+    {
+        public NamingStyleSettingsUpdater(Workspace workspace, string editorconfigPath)
+                : base(workspace, editorconfigPath)
+        {
+        }
+
+        protected override SourceText? GetNewText(
+            SourceText analyzerConfigDocument,
+            IReadOnlyList<((Action<(object, object?)> onSettingChange, NamingStyleSetting option) option, object value)> settingsToUpdate,
+            CancellationToken token)
+        {
+            var result = Parse(analyzerConfigDocument, EditorconfigPath);
+            if (!result.Rules.Any() && settingsToUpdate.Any())
+            {
+                // handle no naming style rules in the editorconfig file.
+                // The implementation does not allow naming style rules to layer meaning all rules are either 
+                // defined in Visual Studios settings or in an editorconfig file. 
+                analyzerConfigDocument = analyzerConfigDocument.WithNamingStyles(Workspace.Options);
```

Comment:
Does this mean that VS naming style rules will automatically be added to editorconfig if there's no existing rules? Seems a little odd... are those updating with the editorconfig UI?

Reply:
So here is where we get the naming style options: https://github.com/dotnet/roslyn/blob/0d68a4bc9378483fc608a79eb963281b158541c7/src/Analyzers/Core/Analyzers/NamingStyle/NamingStyleDiagnosticAnalyzerBase.cs#L182

The implementation of getting the option is this:
https://github.com/dotnet/roslyn/blob/0d68a4bc9378483fc608a79eb963281b158541c7/src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs#L113-L130

Which boils down to:

1. try and get naming styles from the editorconfig
2. if that fails use whatever is in VS

This means that its all or nothing. Either all naming style options come from editorconfig or they all come from VS. Updating a single option will cause the editorconfig have all naming style options persisted to it or else the settings would not persist correctly.